### PR TITLE
Use ffprobe for Metadata Parsing

### PR DESCRIPTION
**Pull request by RLovelett**
Instead of using `ffmpeg` for metadata retrieval use [`ffprobe`](http://www.ffmpeg.org/ffprobe.html).

## [ffprobe Documentation](http://www.ffmpeg.org/ffprobe.html#Description)

> ffprobe gathers information from multimedia streams and prints it in human- and machine-readable fashion.
> For example it can be used to check the format of the container used by a multimedia stream and the format and type of each media stream contained in it.
> If a filename is specified in input, ffprobe will try to open and probe the file content.
> ...
> *ffprobe output is designed to be easily parsable by a textual filter, and consists of one or more sections of a form defined by the selected writer, which is specified by the ‘print_format’ option.*

As far as I am concerned that last line is enough of a reason to use `ffprobe` over `ffmpeg`. Ultimately, I think it would go a long way to make it easier to implement #56 and #11 .

From reading comments on other open pull-requests I gather the project is risk averse to changing the API. For this reason, I tried to minimize changes to the API outputs. Change set 38a5ae2 shows all the ways that the API has changed. In my opinion, none of them are show stopping API changes (mostly changing the units output from kb/s to b/s). Obviously, the maintainers opinion about what is "show stopping" may differ from mine.

I have tested this pull request against `ffmpeg version 1.2.1-static` and `ffmpeg version 2.1.3`. While `2.1.3` provides some additional metadata not found in `1.2.1` nothing is show stopping.

## Notable changes:

* Added `FFMPEG.ffprobe_binary`; similar in nature to `FFMPEG.ffmpeg_binary`
* Both `FFMPEG.ffprobe_binary` and `FFMPEG.ffmpeg_binary` raise `Errno::ENOENT` if the binary cannot be found (see be9b7a2 and d9007d6).
* Add dependency on [`yajl-ruby`](https://rubygems.org/gems/yajl-ruby) for JSON parsing
* Base units from `kb/s` to `b/s`

### CODE REVIEW

**Comment by bjedrocha** [comment]
Any chance of having this pull-request merged?

**Comment by ifeatu** [comment]
+1

**Comment by RLovelett** [comment]
My quick scan says it's been about a year since this repo has seen a committed code change. Is that a sign?

**Comment by dbackeus** [comment]
Since I left Streamio there has been no proper maintainer of the gem :/

Relying on forks is probably the only way to get to any evolution at the moment unless someone receives maintenance authority from Streamio itself (feel free to try and contact them about it).

**Comment by RLovelett** [comment]
Thanks @dbackeus for the info.

**Comment by bjedrocha** [comment]
@RLovelett any chance of you publishing your changes as a gem under a new name?

**Comment by RLovelett** [comment]
Yeah I'll see what I can do. I'll get something setup by next Monday. Any name suggestions?

**Comment by ifeatu** [comment]
@RLovelett ffmpeg-gem??

**Comment by bjedrocha** [comment]
@RLovelett you might as well ask me about [cache invalidation ;)](http://martinfowler.com/bliki/TwoHardThings.html) 

Joking aside, I suggest keeping it simple. How about?

- ruby-ffmpeg
- r-ffmpeg
- rlovelett-ffmpeg
- rlovelett-stremio-ffmpeg

File: lib/ffmpeg/movie.rb [hunk 1]
```diff
@@ -1,10 +1,11 @@
 require 'time'
+require 'multi_json'
 
 module FFMPEG
   class Movie
     attr_reader :path, :duration, :time, :bitrate, :rotation, :creation_time
```
Comment (Good11) [inline lib/ffmpeg/movie.rb]: H

**Comment by codekitchen** [comment]
:+1: any of those names sounds fine to me

**Comment by RLovelett** [comment]
Ok it is done. You can find the source for the fork over [here](https://github.com/RLovelett/rlovelett-ffmpeg) and the published [gem](https://rubygems.org/gems/rlovelett-ffmpeg) here.

I've taken the time to get a CI running and some other stuff. So I think I'm going to update RSpec and some other things and then go through the pull requests on here and possibly merge them into my fork. If you have any that you are interested in seeing merged let me know.

I'll review.

**Comment by bjedrocha** [comment]
Excellent news! Thanks for taking charge @RLovelett.

**Comment by dbackeus** [comment]
@RLovelett many blessings and good luck with the new fork :)

**Comment by codekitchen** [comment]
Thanks much, @RLovelett

**Comment by bikeath1337** [comment]
@RLovelett , I am currently able to manage this project and am planning on merging this after review. It's water under the bridge at this time but I did want to let you know.

**Comment by RLovelett** [comment]
Glad to hear. I'd be happy to help provide help to get the patches I've accepted merged into this repository and delete my fork if this is going to be actively maintained now.

My only goal was to not let this project die. Not have a fork for the sake of it.

**Comment by bikeath1337** [comment]
Superseded by #114. Thanks for the work here, @RLovelett .

**Comment by bikeath1337** [comment]
@RLovelett we're live. https://rubygems.org/gems/streamio-ffmpeg/versions/2.0.0

Thanks to rackfish to publishing!

**Pull request closed without merging**